src/irmin/irmin.mli

```diff
@@ -2257,6 +2268,22 @@ module type S = sig
     val import : t -> slice -> (unit, [ `Msg of string ]) result Lwt.t
     (** [import t s] imports the contents of the slice [s] in [t]. Does
         not modify branches. *)
+
+    val copy :
+      src:t ->
+      dst:t ->
+      ?squash:bool ->
+      ?min:commit list ->
+      ?max:commit list ->
+      unit ->
+      (commit option, [ `Msg of string ]) result Lwt.t
+    (** [copy ~src ~dst ~squash ~min ~max ()] copies the minimal store in the range of
+        commits ([min], [max]) from [src] to [dst]. The commits are exported
+        from [src] and imported in [dst] using a slice, while the nodes and the
+        contents are copied directly from [src] to [dst].
+        If [squash] then Ok Some [c] is returned, where [c] is the commit
+        resulting from merging all commits in the range ([min], [max]). In this
+        case, only [c] (and its subtree) is added to [dst]. *)
```

Comment:
I think `squash` usually means something simpler than that: you just discard all the history and keep the contents of the `max` nodes, e.g. you don't need to merge the contents (as it's already up-to-date in the max nodes!). Not totally sure what the semantics should be when you have multiple disjoint max nodes, maybe just keep all of them?

Reply:
Thanks for clarifying this, `squash` was indeed not clear in my head, I'll fix it. As for the multiple disjoint max nodes, it is simpler to keep them all instead of merging, as you suggest. The user can merge them if he needs it.